src/act_obj.cpp

```diff
@@ -2313,7 +2308,7 @@ void do_hailcorpse(CHAR_DATA *ch, const char *argument) {
             continue;
 
         for (current_obj = current_place->contents; current_obj != nullptr; current_obj = current_obj->next_content) {
-            if ((current_obj->item_type == ITEM_CORPSE_PC) && strstr(current_obj->short_descr, ch->name) != nullptr) {
+            if ((current_obj->item_type == ITEM_CORPSE_PC) && matches_inside(ch->name, current_obj->short_descr)) {
```

Comment:
This is actually interesting, it's a very similar problem to what I faced when working on the summoner - PC corpses don't include the player name in the 'name' field. I made the summoning routine match on short_descr just as this done.